annotation_pipeline/populate_labelstudio.py

```diff
@@ -0,0 +1,243 @@
+        search_term: further refine search with keyword that must be matched in full URL
+        num_pages: number of pages to search (15,000 records per page)
+
+    Returns: 
+        batch_info (pd.Series): summary info of crawl, including filename of csv containing relevant URLs
+    """
+    #run common crawl
+    crawl_return_code, crawl_stdout, crawl_stderr = run_common_crawl(common_crawl_id, url, search_term, num_pages)
+
+    #check success
+    if crawl_return_code != 0:
+        raise ValueError(f"Common crawl script failed:\n{crawl_stderr}")
+
+    #print batch info to verify before continuing
+    batch_info = pd.read_csv("common_crawler/data/batch_info.csv").iloc[-1]
+    print("Batch Info:\n" + f"{batch_info}")
+
+    if(batch_info["Count"] != 0):
+        input("Confirm batch info and csv on HuggingFace, press any key to proceed. Ctrl-c to quit")
+    else:
+        raise ValueError("Batch count is 0. Rerun to crawl more pages.")
+
+    #get urls from hugging face
+    REPO_ID = get_huggingface_repo_id("../common_crawler/config.ini")
+    FILENAME = "urls/" + batch_info["Filename"] + ".csv"
+    df = pd.read_csv(hf_hub_download(repo_id=REPO_ID, filename=FILENAME, repo_type="dataset", local_dir="common_crawler/"), index_col=False)
+
+    return batch_info
+
+def process_tag_collector(batch_info: pd.Series, FILENAME: str) -> str:
+    """
+    Initiates tag collector script and creates a batch id for all samples
+
+    Args:
+        batch_info (pd.Series): summary info for crawl
+        FILENAME (str): filename of csv to collect tags on
+
+    Returns:
+        batch_id (str): a datetime stamp to track batches
+    """
+    
+    #run tag collector
+    tag_collector_return_code, tag_collector_stdout, tag_collector_stderr = run_tag_collector(FILENAME)
+
+    #check success
+    if tag_collector_return_code != 0:
+        raise ValueError(f"Tag collector script failed:\n{tag_collector_stderr}")
+
+    #create batch_id from datetime (removes milliseconds)
+    datetime = batch_info["Datetime"]
+    batch_id = datetime[:datetime.find('.')]
+
+    return batch_id
+
+def label_studio_upload(batch_id: str, FILENAME: str, record_type: str):
+    """
+    Handles label studio task formatting and upload
+    """
+    
+    #convert to label studio task format
+    data = csv_to_label_studio_tasks("labeled-source-text.csv", batch_id, FILENAME, record_type)
+
+    # Load the configuration for the Label Studio API
+    config = LabelStudioConfig("dev.env")
+    if "REPLACE_WITH_YOUR_TOKEN" in config.authorization_token:
+        raise ValueError("Please replace the access token in dev.env with your own access token")
+
+    # Create an API manager
+    api_manager = LabelStudioAPIManager(config)
+
+    #import tasks
+    label_studio_response = api_manager.import_tasks_into_project(data)
+
+    #check import success
+    if label_studio_response.status_code == 201:
```

Comment:
This is a nit (i.e. I won't block on this) but I recommend replacing raw HTTP status code numbers with [HTTPStatus enums](https://docs.python.org/3/library/http.html). This makes things slightly wordier but much more clear. Every time I see a number like this, I have to google which status code it means.